### DISCUSSION TITLE
Some strelka issue's.

### DISCUSSION THREAD

**Discussion by palichx** [Security-Onion-Solutions/securityonion#3416] [Unsupported Versions] [answered]
Hi!
I caught some trouble with strelka and yara. does anybody have same problem?

_1) after restarting (so-strelka-restart) strelka stopping move file from "unprocessing" to "processing"._ 
My case: i wanted to force applay new yara rules - manually restart strelka on sensor.
after that strelka stopping move file. **cause of this** -  duplicate files in folders:

#There are 2 files in both derectory:
Duplicate Filename: SMB-FNHZmk4FYp8RMkQc1a-fb07f730c10748de8bac8feea4a995d4.exe
./processed/SMB-FNHZmk4FYp8RMkQc1a-fb07f730c10748de8bac8feea4a995d4.exe
./unprocessed/SMB-FNHZmk4FYp8RMkQc1a-fb07f730c10748de8bac8feea4a995d4.exe
Duplicate Filename: SMB-FHPEiL3MMex6YvkjJi-52fd5f28e29b83144f64088f9f5251f2.doc
./processed/SMB-FHPEiL3MMex6YvkjJi-52fd5f28e29b83144f64088f9f5251f2.doc
./unprocessed/SMB-FHPEiL3MMex6YvkjJi-52fd5f28e29b83144f64088f9f5251f2.doc

**after removing this files from** "processed" strelka working normally(no need restart). 

_2) so-strelka-restart get ERROR:_ 
[ERROR   ] {'pid': 180688, 'retcode': 1, 'stdout': 'Another process (tcp        0      0 172.17.0.1:33114        172.17.0.5:57314        TIME_WAIT   -                   ) appears to be using port 57314.  Please terminate this process, or reboot to ensure a clean state so that Strelka can start properly

**Looks like need more time before start process**?

_3) Another problem - no log about mistakes in yara rules._ 

2 days ago i updated rules from default repo https://github.com/Neo23x0/signature-base/.
there are 8 new rules:
apt_backdoor_sunburst_fnv1a_experimental.yar
apt_hafnium.yar
apt_ta17_293A.yar
general_cloaking.yar
gen_susp_xor.yar
gen_webshells.yar
thor_inverse_matches.yar
apt_hafnium_log_sigs.yar

Some of them was broken(today's versions is normal). yara scan starting normally after removing this files from manager and manual removing from sensors(4 issuer).
I found in kibana  flag "scan.yara.flags:compiling_error". But i didn't find any log about compiling error. 
**Does anybody know how diagnose this issue?** 

_4) removing yara rules from manager doesn't remove rules from sensors._

in 3rd case i need to remove broken yara files. i try remove it from manager from folder "/opt/so/saltstack/default/salt/strelka/rules/signature-base". but it's not helped me.

**workaround**: remove files from /opt/so/conf/strelka/rules/signature-base/ and /var/cache/salt/minion/files/base/strelka/rules/signature-base/.

Thanks!

**Comment by weslambert** [comment, marked as answer]
Not sure about # 1 at the moment.  We will be improving file moves in the near future.

With regard to # 2, it sounds like something similar to what we are resolving with reserved ports to prevent other system services from using the ports needed by Strelka.

Speaking to # 3, you'll get this error when there is a rule attempting to be loaded that is broken, or that Strelka/YARA can't use.  We are tracking improvement via pre-load rule linting here: https://github.com/Security-Onion-Solutions/securityonion/issues/3419

As for # 4, we'll have to look into this and potentially correct any undesired behavior.

**Reply by palichx** [reply]
Thank you for your answres!

2) in my sensors i dont't have any services on this port.  after stopping strelka in system remains 1 connetcion in "TIME_WAIT" state.

#netstat -an | grep 57314
tcp        0      0 172.17.0.1:**41540**        172.17.0.5:57314        ESTABLISHED
tcp6       0      0 :::57314                :::*                    LISTEN
tcp6       0      0 10.1.115.163:57314      172.17.0.8:52388        ESTABLISHED

#so-strelka-stop
..
#netstat -an | grep 57314
tcp        0      0 172.17.0.1:**41540**        172.17.0.5:57314        TIME_WAIT

#so-strelka-start
..
[ERROR   ] Command 'netstat' failed with return code: 1
[ERROR   ] stdout: Another process (tcp        0      0 172.17.0.1:**41540**        172.17.0.5:57314        TIME_WAIT   -                   ) appears to be using port 57314.  Please terminate this process, or reboot to ensure a clean state so that Strelka can start properly.
[ERROR   ] retcode: 1
[ERROR   ] {'pid': 4177, 'retcode': 1, 'stdout': 'Another process (tcp        0      0 172.17.0.1:**41540**        172.17.0.5:57314        TIME_WAIT   -                   ) appears to be using port 57314.  Please terminate this process, or reboot to ensure a clean state so that Strelka can start properly.', 'stderr': ''}

in a minute this connection will be close, and strelka will start without errors.

**Reply by weslambert** [reply]
Yeah, I need to look into the port issue.  It seems like something with Docker, and Strelka eventually is fine, but it is certainly not ideal.  I've created an issue for tracking this here: https://github.com/Security-Onion-Solutions/securityonion/issues/3457

**Reply by weslambert** [reply]
Port "conflict" is resolved in the aforementioned issue.

**Reply by palichx** [reply]
Thanks!
